plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts

```diff
@@ -306,6 +320,17 @@ export const createPublishGithubPullRequestAction = (
             {
               files,
               commit: commitMessage ?? title,
+              author:
```

Comment:
the structure of both author and comitter looks the same so let's declare it above instead of repeating the same ternary operators in two places

Reply:
They are similar but the difference is author is required to have both email and name while committer can support one of them.  If you have any suggestions on how to handle this let me know.

Reply:
@jhaals Is there additional thoughts you can share on how to handle difference is author is required to have both email and name while committer

Reply:
@jhaals I have extracted reduce the complexity on the response please review.